feat: Add LLM A2A Integration module

hungpq-ai opened the pull request:
Add Agent-to-Agent (A2A) protocol integration for LLM, enabling communication between AI agents using Google's A2A protocol.

Features:
- Connect to external A2A agents
- Delegate tasks to specialized agents
- Agent discovery via Agent Cards
- Streaming support for real-time responses
- Multi-agent orchestration within LLM chat

Author by Hung Pham Quang (VietNam)

hungpq-ai committed:
feat: Add LLM A2A Integration module

Add Agent-to-Agent (A2A) protocol integration for LLM, enabling communication
between AI agents using Google's A2A protocol.

Features:
- Connect to external A2A agents
- Delegate tasks to specialized agents
- Agent discovery via Agent Cards
- Streaming support for real-time responses
- Multi-agent orchestration within LLM chat

Author by Hung Pham Quang (VietNam)

adar2378 commented:
@hungpq-ai Thanks for the contribution, we will check and get back to you.

adar2378 reviewed (commented):
@hungpq-ai 

## Testing Feedback

Thanks for this module! I tested it with the [Google ADK hello_world_agent](https://google.github.io/adk-docs/a2a/quickstart-exposing/) and found an issue.

### Setup ✅
- A2A agent running at `http://localhost:9999`
- Created `llm.a2a.agent` record, test connection successful
- Assigned agent to thread (GPT-4o)

### Issue ❌
**Agent delegation doesn't work** - LLM acknowledges it will use the agent ("I'll ask the agent...") but never makes the actual tool call. No HTTP request reaches the A2A agent server.

**Logs:**
```
INFO: Added A2A capability prompt: 1 agents
INFO: Added 1 A2A delegation tools to thread 1
INFO: Storing 1 A2A tools for provider 1
INFO: POST https://api.openai.com/v1/chat/completions "HTTP/1.1 200 OK"
# No tool call, no request to A2A agent ❌
```

### Questions
1. Have you tested end-to-end delegation successfully?
2. Which LLM model did you test with?

### Suggestion
Add a README with:
- Quick start guide (how to set up an A2A agent)
- Example agent configuration
- Expected behavior

---

**Test env:** Odoo 18.0, llm_openai, gpt-4o, Google ADK v1.21.0

hungpq-ai commented:
Okay, I’ll get back to you soon. I’m a bit busy at work right now.

ayushin commented:
@hungpq-ai looking forward to the fixed mergeable code

hungpq-ai committed:
feat: Update LLM A2A module with README and dependency enhancements
- Added llm_openai as a dependency to ensure proper method resolution order.
- Introduced a comprehensive README.md detailing features, installation, quick start, and troubleshooting for the LLM A2A integration.

hungpq-ai commented:
> @hungpq-ai looking forward to the fixed mergeable code

I’ve updated my commit. Please have a look and let me know if you find any issues.
